components/locid/src/subtags/language.rs

```diff
@@ -198,8 +194,33 @@ impl Language {
     ///
     /// assert_eq!(lang.is_empty(), true);
     /// ```
+    #[inline]
     pub fn is_empty(self) -> bool {
-        self.0.is_none()
+        self == UND
+    }
+
+    /// Returns an `Option<Language>`, which is `None` if the subtag is `"und"`,
+    /// and `Some(self)` otherwise.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu::locid::subtags::Language;
+    ///
+    /// assert!(matches!(
+    ///     Language::und().to_option(),
+    ///     None));
+    /// assert!(matches!(
+    ///     Language::from_bytes(b"uk").unwrap().to_option(),
+    ///     Some(_)));
+    /// ```
+    #[inline]
+    pub fn to_option(self) -> Option<Language> {
```

Comment:
thought: perhaps we should instead just be checking `is_empty()` where this is used? I don't mind either way

Reply:
yes, I'm opinionated here - this should use `is_empty`. `und` is just the way we stringify empty, but if you ignore stringification, the data model of `LanguageIdentifier` is empty/not-empty.